sphinx_exercise/post_transforms.py

```diff
@@ -0,0 +1,274 @@
+import sphinx.addnodes as sphinx_nodes
+from sphinx.transforms.post_transforms import SphinxPostTransform
+from sphinx.util import logging
+from sphinx.builders.latex import LaTeXBuilder
+from docutils import nodes as docutil_nodes
+
+from .utils import get_node_number, get_refuri, find_parent
+from .nodes import (
+    exercise_enumerable_node,
+    solution_node,
+    exercise_title,
+    exercise_subtitle,
+    solution_title,
+    solution_subtitle,
+)
+
+logger = logging.getLogger(__name__)
+
+
+# Resolver Functions
+
+
+def resolve_enumerated_exercise_node_title(app, node):
+    title = node.children[0]
+    if isinstance(title, exercise_title):
+        node_number = get_node_number(app, node, "exercise")
+        updated_title = exercise_title()
+        if title.default_title():
+            title_text = title.children[0].astext() + f" {node_number}"
+            updated_title += docutil_nodes.Text(title_text)
+        else:
+            updated_title += title.children[0]
+        updated_title += title.children[1:]
+        node.children[0] = updated_title
+    node["title"] = title_text  # TODO: Include subtitle text?
+    node.resolved_title = True
+    return node
+
+
+def build_reference_node(app, target_node):
+    refuri = app.builder.get_relative_uri(
+        app.env.docname, target_node.get("docname", "")
+    )
+    refuri += "#" + target_node.get("label")
+    reference = docutil_nodes.reference(
+        "",
+        "",
+        internal=True,
+        refuri=refuri,
+        anchorname="",
+    )
+    return reference
+
+
+def resolve_solution_node_title(app, node):
+    """ Resolve Solution Nodes """
+    exercise_target = app.env.sphinx_exercise_registry[node.get("target_label")]["node"]
+    title = node.children[0]
+    if isinstance(title, solution_title):
+        new_title = solution_title()
```

Comment:
@AakashGfude similarly here?

Reply:
similarly for this one.

Reply:
> adding visit, depart exercise_title support for latex here:

@AakashGfude indeed. All the `LaTex` elements are current `raise NotImplementedError` for title and subtitle visit/depart.